llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -4140,6 +4266,165 @@ bool AIE2InstructionSelector::select512BitG_AIE_STORE_SRS(
   }
 }
 
+// Returns the instruction where the Reg is defined ignoring G_BITCAST.
+static MachineInstr *stripBitCast(Register Reg, MachineRegisterInfo &MRI) {
+  MachineInstr *MI = getDefIgnoringCopies(Reg, MRI);
+  if (MI->getOpcode() == TargetOpcode::G_BITCAST)
+    return getDefIgnoringCopies(MI->getOperand(1).getReg(), MRI);
+  return MI;
+}
+
+// If two 256-bit SRS are input to 512-bit STORE through CONCAT, these are
+// combined into VST.SRS
+bool AIE2InstructionSelector::selectG_AIE_STORE_CONCAT_SRS(
+    MachineInstr &StoreI, MachineRegisterInfo &MRI) {
+
+  Register ConcatResult = (StoreI.uses().begin())->getReg();
+  MachineInstr *ConcatOp = MRI.getUniqueVRegDef(ConcatResult);
+
+  assert(ConcatOp && "Expected SSA.");
+
+  if (getLoadStoreSize(StoreI) != 512 ||
+      ConcatOp->getOpcode() != AIE2::G_INTRINSIC ||
+      cast<GIntrinsic>(*ConcatOp).getIntrinsicID() !=
+          Intrinsic::aie2_concat_I512_I256)
+    return false;
+
+  MachineInstr *SrsOpLo = stripBitCast(ConcatOp->getOperand(2).getReg(), MRI);
+  MachineInstr *SrsOpHi = stripBitCast(ConcatOp->getOperand(3).getReg(), MRI);
+
+  if (!canCombineSRSUPS(StoreI, *SrsOpLo) ||
+      !canCombineSRSUPS(StoreI, *SrsOpHi))
+    return false;
+
+  std::optional<AddressingModeInfo> AMI =
+      getOrDefineAddressingRegister(StoreI, MRI);
+  if (!AMI)
+    return false;
+
+  Register SrcRegLo = SrsOpLo->getOperand(2).getReg();
+  Register ShftRegLo = SrsOpLo->getOperand(3).getReg();
+  Register SignRegLo = SrsOpLo->getOperand(4).getReg();
```

Comment:
Nit: You could use something like `auto [Res, CarryOut, LHS, RHS, CarryIn] = MI.getFirst5Regs();`

Reply:
Unable to use `getFirst5Regs` as second operand is not reg(intrinsic)